pkg/api/types.go

```diff
@@ -518,8 +518,18 @@ type GitRepoVolumeSource struct {
 // The contents of the target Secret's Data field will be presented in a volume
 // as files using the keys in the Data field as the file names.
 type SecretVolumeSource struct {
-	// Name of the secret in the pod's namespace to use
-	SecretName string `json:"secretName"`
+	// Name of the secret in the pod's namespace to use.  May not be specified if items is specified.
+	SecretName string `json:"secretName,omitempty"`
+	// A list of projections to make into the volume.  May not be specified if SecretName is specified.
+	Items []SecretVolumeFile `json:"items,omitempty"`
+}
+
+// SecretVolumeFile represents a single secret key projected into a file.
+type SecretVolumeFile struct {
+	// A reference to the secret key to project into the volume.
+	SecretKeySelector `json:",inline"`
```

Comment:
You're right that this looks like an API break.  I think my proposal avoids that, too.

I do think we should use the same model for ConfigMap and Secrets, and I think the linked proposal should be changed.

If we eventually need to compose multiple ConfigMaps or Secrets into a single directory, we could pluralize and have a list of {name, items[]}, leaving the singular one as the first instance.  It's a little clunky but it gives us a compatible API that can be fixed in v2.